modules/nf-core/fastqc/main.nf

```diff
@@ -25,6 +25,11 @@ process FASTQC {
     def old_new_pairs = reads instanceof Path || reads.size() == 1 ? [[ reads, "${prefix}.${reads.extension}" ]] : reads.withIndex().collect { entry, index -> [ entry, "${prefix}_${index + 1}.${entry.extension}" ] }
     def rename_to = old_new_pairs*.join(' ').join(' ')
     def renamed_files = old_new_pairs.collect{ old_name, new_name -> new_name }.join(' ')
+
+    def memory_in_mb = MemoryUnit.of("${task.memory}").toUnit('MB')
+    // FastQC memory value allowed range (100 - 10000)
+    def fastqc_memory = memory_in_mb > 10000 ? 10000 : (memory_in_mb < 100 ? 100 : memory_in_mb)
```

Comment:
Did you mean this ?
```suggestion
    def fastqc_memory = memory_in_mb > 10000 ? 10000 : (memory_in_mb > 100 ? 100 : memory_in_mb)
```

Reply:
FastQC will give an error if `--memory` is set outside outside the allowed range (100 - 10000). I meant if task.memory is greater than 10gb, set `--memory 10000`, if task.memory is less than 100 MB, set `--memory 100`.

Reply:
So yes (?) 😄

Reply:
I see, sorry for not understanding your comment above

Reply:
Not at all, bit of a strange behaviour from FastQC.